src/components/builds/BuildDetails.tsx

```diff
@@ -223,13 +223,20 @@ function BuildDetails(props: Props) {
   );
 
   const hasWritePermission = hasWritePermissions(build.repository.viewerPermission);
+  const allTaskIds = build.latestGroupTasks.map(task => task.id);
   const failedTaskIds = build.latestGroupTasks
     .filter(task => task.status === 'FAILED' || (task.status === 'ABORTED' && task.requiredGroups.length === 0))
     .map(task => task.id);
   const runningTaskIds = build.latestGroupTasks
     .filter(task => ['SCHEDULED', 'CREATED', 'EXECUTING', 'TRIGGERED'].includes(task.status))
     .map(task => task.id);
   const reRunAllTasksButton =
+    allTaskIds.length === runningTaskIds.length || !hasWritePermission ? null : (
```

Comment:
Should we show the "Re-run all tasks" only if nothing is running aka `runningTaskIds.length === 0`? What do you think? 🤔

Reply:
Agree.
Also just thinking may be we need "Re-run build" button instead of this? Independent of `runningTaskIds.length`?

Reply:
IMO "Re-Run All Tasks" looks good and consistent with GitHub Actions 👌

<img width="1496" alt="Screenshot 2022-11-16 at 1 31 47 PM" src="https://user-images.githubusercontent.com/989066/202143412-923b7126-b9cf-417e-9d13-066f507c4d19.png">